server/service/src/requisition/request_requisition/insert_program.rs

```diff
@@ -99,60 +142,49 @@ fn validate(
         .store_id()
         .ok_or(OutError::OtherPartyIsNotAStore)?;
 
-    Ok(())
+    // Check if the 'other_party' is a valid Program Supplier
+    program_settings
```

Comment:
this should be for program_setting for the order type (since there are different suppliers for different programs)